shotgun_api3/shotgun.py

```diff
@@ -2365,10 +2365,27 @@ def _upload_to_sg(self, entity_type, entity_id, path, field_name, display_name,
 
         params.update(self._auth_params())
 
+        # If we ended up with a unicode string path, we need to encode it
+        # as a utf-8 string. If we don't, there's a chance that there will
+        # will be an attempt later on to encode it as an ascii string, and
+        # that will fail ungracefully if the path contains any non-ascii
+        # characters.
+        #
+        # On Windows, if the path contains non-ascii characters, the calls
+        # to open later in this method will fail to find the file if given
+        # a utf-8 encoded string path. In that case, we're going to have
+        # to call open on the unicode path, but we'll use the utf-8 encoded
+        # string for everything else.
+        path_to_open = path
+        if isinstance(path, unicode):
```

Comment:
i am guessing if someone now uploads a path on windows and passes a utf-8 encoded string to the API, it will still fail, right? Maybe we can add a test for that and add a little bit more logic. Would be great if we tested in our unit tests upload with ascii string, utf-8 encoded string (with special characters), unicode only containing 8 bit ascii chars and unicode containing special characters). If those tests pass we should be ok (i think!)

Reply:
on a side note, what happens if you pass a `shift-jis` encoded string to the API - might be worth adding a test for that too. This is not uncommon as a locale still in japan for example. Would be nice to validate that if the input is not utf-8 and is not unicode, we provide a sensible error message.

Reply:
Good things to check. Thinking about it, you're probably right about getting a utf-8 encoded string on Windows and it failing.